### PR TITLE
Tests for entitySelect.js and Modification of Storage

### PR DESCRIPTION
**Pull request by varunsivashankar**
**entitySelect.js**
- This tests the function _identifyEntityClass_. It's functionality was described in an earlier pull request.

image

**entitySelectContent.js**
- Modified the _selectClick_ function so that it works for the new method of storage for OUs. (This function is not designed to work for users and groups because they do not appear on the DOM of the settings page).
image

**preferred.js**
- The data about which select button was most recently clicked is now stored in _chrome.storage.local_ instead of _chrome.storage.sync_.
- This was necessary to avoid issues that might arise if the same account is being used on two different devices.
- Tests in preferredTest.js had to be modified to accommodate this modified implementation.

image

**Generalization**
- The Add/Remove button's text is now dependent on whether the admin is on an OU, user or group page. I made some changes in the function _findPageType_ in popup.js to implement this.
- Some tests for popup.js had to be modified.
image

### CODE REVIEW

File: STEP_Project/tests/entitySelectTest.js [hunk 1]
```diff
@@ -1,71 +1,29 @@
...
         storageObj = mockChrome;
-    })
-
-    afterAll(function()
-    {
-        tabl.remove();
-    })
+    });
     
-    it('Should update the mock storage if entity\'s name is changed', ()=>
```
Comment (atishey-jain) [inline STEP_Project/tests/entitySelectTest.js:50]: Why are we deleting this test ?

File: STEP_Project/popup.js [hunk 2]
```diff
@@ -27,9 +27,30 @@ function findPageType(givenurl)
...
     pageType = PAGE_TYPES.OTHER;
+
     if (entitypage !== null)
     {
         pageType = PAGE_TYPES.ORG_LIST_PAGE;
+
+        let entityPageOU = givenurl.match(/admin.google.com\/(u\/[0-9]\/)?ac\/orgunits/g);
+        let entityPageUser = givenurl.match(/admin.google.com\/(u\/[0-9]\/)?ac\/users/g);
+        let entityPageGroup = givenurl.match(/admin.google.com\/(u\/[0-9]\/)?ac\/groups/g);
+
```
Comment (atishey-jain) [inline STEP_Project/popup.js:38]: We can reuse findPageType function of popup.js here. Consider moving that function to a shared js file, say common.js and move its tests there. Then. use that function here and where it is originally being used.

File: STEP_Project/tests/entitySelectTest.js (hunk 1 above)
Comment (varunsivashankar) [reply STEP_Project/tests/entitySelectTest.js:50]: This was not a test for this file. It's part of another test file. Included it here initially just for syntax.

File: STEP_Project/popup.js (hunk 2 above)
Comment (varunsivashankar) [reply STEP_Project/popup.js:38]: This is part of the same function (findPageType). I've just increased the functionality.

**Pull request merged**